Main.java

```diff
@@ -109,51 +96,97 @@ static void btnCrudGoBackMethod (){
     }
     static void btnCrudDeleteMethod (){
-
+        String result="Waiting access...";
+        try{
+            result=db1.delete(conn); //function passes result onto the string
+        }catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        crudOutputStream.push(result); //send result to stream for render later
     }
-    static void simRender(){
-        if (simulationRunning==true) {
+    static void appRender() {
+
+        txtMainStatusBar.setText(txtMainStatusBarOutput); //update the status bar of main frame
+
+        if (crudRunning == true) {
+            crudRenderTest++;
+            if (crudRenderTest>10) crudRenderTest=0;
+            crudOutputStream.push(Integer.toString(crudRenderTest));
+            txtCrudOutput.setText(crudOutputStream.getStream());
+        }
+        if (simulationRunning == true) {
             timeSeconds++; //add time
-            txtSimTime.setText("Time="+Integer.toString(timeSeconds)); //print the time
+            txtSimTime.setText("Time=" + Integer.toString(timeSeconds)); //print the time
 
             //process the time
-            if (timeSeconds >= 86400){
+            if (timeSeconds >= 86400) {
                 simOutputStream.push("Day Over!-------------------------------------------");
                 simulator01.resetAllSuppliers();
-                timeSeconds=0;
+                timeSeconds = 0;
                 dayCounter++;
-                if (dayCounter<7) simOutputStream.push("Starting day "+dayCounter);
+                if (dayCounter < 7) simOutputStream.push("Starting day " + dayCounter);
             }
             //...advance day counter each time the seconds hits 86400, then reset the seconds
             //...if day reaches 6, then simulation is over. This program simulates a week at a time.
-            if (dayCounter > 6) simulationRunning=false;
+            if (dayCounter > 6) simulationRunning = false;
 
             //finalize and render all result at the end of the frame
             txtSimOutput.setText(simOutputStream.getStream());
             txtSimOutput2.setText(simulator01.printTotalResult());
         }
-        if(crudRunning==true){
-            txtCrudOutput.setText(crudOutputStream.getStream());
-        }
     }
 
+
     //PROGRAM ENTRY POINT----------------------------
     public static void main(String[] args) {
 
         GUIInit(); //initialize the GUI, buttons, actionListeners, labels, jpanels, etc.
 
         simulator01.initializeSimulatorData(); //initialize the simulator
+        timer.start(); //start rendering
+
+        //Establish connection to database
+        try {
+
+            conn = DriverManager.getConnection(dbURL, dbUsername, dbPassword);
+
+            if (conn != null) {
+                //report status to the status bar in the main frame
+                txtMainStatusBarOutput="Status: ******CONNECTING******";
+                txtMainStatusBarOutput="Status: Connected to database";

```

Comment:
should this be:
 else { txtMainStatusBarOutput= "status:connected to database?";}?

Reply:
Do you mean add an else statement? When status bar reads empty, it already implies connection hasn't been established yet. Actually the connection happens so fast that the bar never gets a chance to display "connecting"